proctor-common/src/main/java/com/indeed/proctor/common/ProctorUtils.java

```diff
@@ -1417,4 +1440,29 @@ private static boolean evaluatePayloadValidator(
 
         return false;
     }
+
+    private static boolean isUnitlessAllocation(@Nonnull final Allocation allocation) {
+        final int unitlessIdentifierIndex =
+                allocation.getRule().indexOf(UNITLESS_ALLOCATION_IDENTIFIER);
+        final boolean unitlessButFalse =
+                allocation.getRule().contains(UNITLESS_ALLOCATION_IDENTIFIER + " == false");
+        return !unitlessButFalse
+                && unitlessIdentifierIndex != -1
+                && (unitlessIdentifierIndex == 0
+                        || allocation.getRule().charAt(unitlessIdentifierIndex - 1) != '!');
+    }
+
+    public static boolean containsUnitlessAllocation(@Nonnull final TestDefinition testDefinition) {
+        return testDefinition.getEnableUnitlessAllocations()
+                && testDefinition.getAllocations().stream()
+                        .anyMatch(
+                                (allocation) ->
+                                        allocation.getRule() != null
+                                                && isUnitlessAllocation(allocation)
+                                                && allocation.getRanges().stream()
+                                                        .anyMatch(
+                                                                range ->
+                                                                        range.getLength()
+                                                                                > 0.9999));
```

Comment:
nit: should this 100% check be extracted to a method?

Reply:
I think it is fine as is.